src/api/cpp/array.cpp

```diff
@@ -250,9 +242,50 @@ INSTANTIATE(__half);
 #undef INSTANTIATE
 
 array::~array() {
-    af_array tmp = get();
+#ifdef AF_UNIFIED
+    using af_release_array_ptr =
+        std::add_pointer<decltype(af_release_array)>::type;
+    static auto &instance = unified::AFSymbolManager::getInstance();
+
+    if (get()) {
+        af_backend backend = instance.getActiveBackend();
+        af_err err         = af_get_backend_id(&backend, get());
+        if (!err) {
+            switch (backend) {
+                case AF_BACKEND_CPU: {
+                    static auto cpu_handle = instance.getHandle();
+                    static af_release_array_ptr func =
+                        reinterpret_cast<af_release_array_ptr>(
+                            common::getFunctionPointer(cpu_handle,
+                                                       "af_release_array"));
+                    func(get());
+                    break;
+                }
+                case AF_BACKEND_OPENCL: {
+                    static auto opencl_handle = instance.getHandle();
+                    static af_release_array_ptr func =
+                        reinterpret_cast<af_release_array_ptr>(
+                            common::getFunctionPointer(opencl_handle,
+                                                       "af_release_array"));
+                    func(get());
+                    break;
+                }
+                case AF_BACKEND_CUDA: {
+                    static auto cuda_handle = instance.getHandle();
+                    static af_release_array_ptr func =
```

Comment:
Are these statics helping that much ? Is `getFunctionPointer` that costly even now ?

Reply:
Yes, I experimented with removing the static variables and it helps the runtime.